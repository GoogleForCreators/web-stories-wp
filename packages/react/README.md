# React

Generic package that re-exports parts of `react` and `react-dom` alongside some other helpful utils and hooks.
