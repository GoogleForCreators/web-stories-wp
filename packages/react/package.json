{
  "name": "@googleforcreators/react",
  "description": "A tiny wrapper package that re-exports React and provides additional utilities and custom hooks.",
  "private": false,
  "version": "0.1.202204141511",
  "author": {
    "name": "Google",
    "url": "https://creators.google/"
  },
  "license": "Apache-2.0",
  "keywords": [
    "web stories"
  ],
  "homepage": "https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/react/README.md",
  "repository": {
    "type": "git",
    "url": "https://github.com/GoogleForCreators/web-stories-wp.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/googleforcreators/web-stories-wp/issues"
  },
  "engines": {
    "node": ">= 12 || >= 14 || >= 16",
    "npm": ">= 7.3"
  },
  "customExports": {
    ".": {
      "default": "./src/index.js"
    }
  },
  "main": "dist/index.js",
  "module": "dist-module/index.js",
  "source": "src/index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "prop-types": "^15.8.1",
    "resize-observer-polyfill": "^1.5.1",
    "shallow-equal": "^1.2.1",
    "use-context-selector": "^1.2.11",
    "use-debounce": "^7.0.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.5",
    "@testing-library/react-hooks": "^8.0.0"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
