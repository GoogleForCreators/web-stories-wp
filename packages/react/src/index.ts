/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export { useDebouncedCallback } from 'use-debounce';
export { default as useReduction } from 'use-reduction';

export * from './context';
export * from './react';
export * from './react-dom';

export { default as useBatchingCallback } from './useBatchingCallback';
export { default as useCombinedRefs } from './useCombinedRefs';
export { default as useContextSelector } from './useContextSelector';
export { default as useFocusOut } from './useFocusOut';
export { default as usePrevious } from './usePrevious';
export { default as useResizeEffect } from './useResizeEffect';
export { default as renderToStaticMarkup } from './renderToStaticMarkup';
export { default as useUnmount } from './useUnmount';
export { default as useWhyDidYouUpdate } from './useWhyDidYouUpdate';
export { default as useInitializedValue } from './useInitializedValue';
export { default as shallowEqual } from './shallowEqual';
