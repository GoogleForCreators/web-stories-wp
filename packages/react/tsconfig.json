{
  "extends": "../../tsconfig.shared.json",
  "compilerOptions": {
    "rootDir": "src",
    "declarationDir": "dist-types"
  },
  "include": ["src/context.ts", "src/react.ts", "src/react-dom.ts"]
}
