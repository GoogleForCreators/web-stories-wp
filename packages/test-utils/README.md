# Test Utils

Package for Web Stories test utilities
