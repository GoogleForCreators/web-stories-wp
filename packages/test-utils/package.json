{
  "name": "@googleforcreators/test-utils",
  "description": "Package for Web Stories test utilities",
  "private": true,
  "version": "0.1.202203150806",
  "author": {
    "name": "Google",
    "url": "https://creators.google/"
  },
  "license": "Apache-2.0",
  "keywords": [
    "web stories"
  ],
  "homepage": "https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/test-utils/README.md",
  "repository": {
    "type": "git",
    "url": "https://github.com/GoogleForCreators/web-stories-wp.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/googleforcreators/web-stories-wp/issues"
  },
  "engines": {
    "node": ">= 12 || >= 14 || >= 16",
    "npm": ">= 7.3"
  },
  "customExports": {
    ".": {
     "default": "./src/index.js"
    }
  },
  "main": "dist/index.js",
  "module": "dist-module/index.js",
  "source": "src/index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {},
  "devDependencies": {
    "@googleforcreators/design-system": "*",
    "@testing-library/react": "^12.1.5",
    "prop-types": "^15.8.1",
    "styled-components": "^5.3.3"
  }
}
