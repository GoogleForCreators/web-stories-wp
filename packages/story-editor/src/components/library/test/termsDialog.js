/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { waitFor, fireEvent, screen } from '@testing-library/react';
import {
  localStore,
  LOCAL_STORAGE_PREFIX,
} from '@googleforcreators/design-system';
import { renderWithTheme } from '@googleforcreators/test-utils';

/**
 * Internal dependencies
 */
import TermsDialog from '../panes/media/media3p/termsDialog';

jest.mock('@googleforcreators/design-system', () => ({
  __esModule: true,
  ...jest.requireActual('@googleforcreators/design-system'),
  localStore: {
    setItemByKey: jest.fn(),
    getItemByKey: jest.fn(() => false),
  },
}));

describe('TermsDialog', () => {
  it('should store the acknowledgement in the localStore when the window is dismissed', async () => {
    renderWithTheme(<TermsDialog />);
    const dismiss = screen.queryByText(/^Dismiss/);
    expect(localStore.getItemByKey).toHaveBeenCalledWith(
      LOCAL_STORAGE_PREFIX.TERMS_MEDIA3P
    );

    fireEvent.click(dismiss);

    await waitFor(() =>
      expect(localStore.setItemByKey).toHaveBeenCalledWith(
        LOCAL_STORAGE_PREFIX.TERMS_MEDIA3P,
        true
      )
    );
  });
});
