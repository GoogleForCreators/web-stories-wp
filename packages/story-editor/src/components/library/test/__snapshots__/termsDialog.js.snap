// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsDialog should render 1`] = `
.c1 {
  width: 12px;
  margin-left: 0.5ch;
  margin-bottom: 2px;
  stroke-width: 0;
  vertical-align: text-bottom;
}

.c0 {
  font-family: 'Google Sans',sans-serif;
  color: #131516;
  margin: 0;
  padding: 0;
  font-family: 'Google Sans',sans-serif;
  font-size: 14px;
  font-weight: 700;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  line-height: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4F7ED9;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  vertical-align: baseline;
}

.c0:focus {
  box-shadow: none;
}

.c0:hover {
  color: #476ABD;
}

.c0:focus {
  color: #4F7ED9 !important;
}

.c0.focus-visible,
.c0[data-focus-visible-added] {
  outline: none;
  box-shadow: 0px 0px 0 2px #FFF,0px 0px 0 4px #5D96EB;
}

<a
  aria-label="Learn more by visiting Web Stories for WordPress"
  class="c0"
  href="https://wp.stories.google/docs/terms/"
  rel="noreferrer"
  target="_blank"
>
  <span>
    Learn more.
    <span
      class="c1"
    />
  </span>
</a>
`;
